controllers/mlmd_test.go

```diff
@@ -0,0 +1,249 @@
+//go:build test_all || test_unit
+// +build test_all test_unit
+
+package controllers
+
+import (
+	"testing"
+
+	dspav1alpha1 "github.com/opendatahub-io/data-science-pipelines-operator/api/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func TestDeployMLMD(t *testing.T) {
+	testNamespace := "testnamespace"
+	testDSPAName := "testdspa"
+	expectedMLMDEnvoyName := "ds-pipeline-metadata-envoy-testdspa"
+	expectedMLMDGRPCName := "ds-pipeline-metadata-grpc-testdspa"
+	expectedMLMDWriterName := "ds-pipeline-metadata-writer-testdspa"
+
+	// Construct MLMD Spec
+	dspa := &dspav1alpha1.DataSciencePipelinesApplication{
+		Spec: dspav1alpha1.DSPASpec{
+			APIServer: &dspav1alpha1.APIServer{
+				ArchiveLogs: true, // TODO: This appears to be required which is out-of-spec (.Spec.APIServer should be fully defaultable), but test throws an nil pointer panic if it isn't provided. possibly due to test setup - Investigate.
```

Comment:
can we multi-line this comment (and not have it be inline)

Reply:
also should this not be a gh issue instead of a todo comment?

Reply:
same with the other 2 areas where this shows up (though I don't think we need to repeat the same comment multiple times)

Reply:
So for now I'm leaving it as a TODO because it's unclear if this is due to using a 'minimally viable spec' and not what we'd see on a running system (where omitted fields are expanded/defaulted, etc)